### PR TITLE
User Can Fetch All Meals

### PR DESCRIPTION
**Pull request by alexander-mathieu**
This PR adds an endpoint for a user to fetch all meals, along with their corresponding foods.

- Changes beforeCreate hook in Food/Meal models to beforeSave hook
- Changes beforeEach test blocks to asynchronous functions
- Adds spec for user fetching all meals and foods
- Adds mealsRouter with index route

### CODE REVIEW

File: tests/api/v1/foods_api_endpoint.spec.js [hunk 1]
```diff
@@ -7,8 +7,8 @@ var app = require('../../../app');
 var cleanup = require('../../../tests/helpers/test_clear_database');
 
 describe('foods api endpoint', () => {
-  beforeEach(() => {
-    cleanup();
+  beforeEach(async function() {
```
Comment (CosmicSpagetti) [inline tests/api/v1/foods_api_endpoint.spec.js]: This is great no more failing test.

File: tests/api/v1/meals_api_endpoint.spec.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+var shell = require('shelljs');
+var request = require('supertest');
+
+var Meal = require('../../../models').Meal;
+var Food = require('../../../models').Food;
```
Comment (CosmicSpagetti) [inline tests/api/v1/meals_api_endpoint.spec.js]: Hey are we using this?!!

File: tests/api/v1/meals_api_endpoint.spec.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      return request(app)
+      .get('/api/v1/meals')
+      .then(response => {
+        expect(response.statusCode).toBe(200);
+
+        expect(response.body.length).toBe(2);
+        expect(Object.keys(response.body[0])).toContain('id');
+        expect(Object.keys(response.body[0])).toContain('name');
+        expect(Object.keys(response.body[0])).toContain('foods');
+
```
Comment (CosmicSpagetti) [inline tests/api/v1/meals_api_endpoint.spec.js]: ## SO ROBUST

File: tests/api/v1/meals_api_endpoint.spec.js (hunk 2 above)
Comment (alexander-mathieu) [reply tests/api/v1/meals_api_endpoint.spec.js]: Probably not, I'll remove it.  Good catch. ⚾️

**Pull request merged**